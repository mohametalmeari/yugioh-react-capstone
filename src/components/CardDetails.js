import PropTypes from 'prop-types';
import { ArrowIcon } from '../icons';

const CardDetails = ({ cardsRace, cardsCount }) => (
  <div className="card-details">
    <h3 className="cards-title">
      {/* &quot;
      {'Level '} */}
      {cardsRace}
      {/* &quot;
      {' Monster'} */}
    </h3>
    <p>
      {cardsCount}
      {' '}
      Cards
    </p>
    <ArrowIcon />
  </div>
);
CardDetails.propTypes = {
  cardsRace: PropTypes.string.isRequired,
  cardsCount: PropTypes.string.isRequired,
};

export default CardDetails;
