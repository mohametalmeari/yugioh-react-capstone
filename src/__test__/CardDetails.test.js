import { render, screen, cleanup } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import CardDetails from '../components/CardDetails';

describe('Cards Details Component', () => {
  afterEach(() => cleanup());
  test('renders race', () => {
    render(
      <MemoryRouter>
        <CardDetails
          key={888}
          cardsRace="Continuous"
          cardsCount={20}
        />
      </MemoryRouter>,
    );
    expect(screen.getByText(/Continuous/i)).toBeInTheDocument();
    expect(screen.getByText(/20 cards/i)).toBeInTheDocument();
  });
  test('renders Race Snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <CardDetails
          key={888}
          cardsRace="Continuous"
          cardsCount={20}
        />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
