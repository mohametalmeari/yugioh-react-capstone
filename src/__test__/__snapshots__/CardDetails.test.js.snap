// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards Details Component renders Race Snapshot 1`] = `
<div>
  <div
    class="card-details"
  >
    <h3
      class="race-title"
    >
      Continuous
    </h3>
    <p>
      20
       
      cards
    </p>
    <svg
      class="w-6 h-6"
      fill="none"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12.75 15l3-3m0 0l-3-3m3 3h-7.5M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  </div>
</div>
`;
